sound/soc/sof/loader.c

```diff
@@ -358,6 +359,86 @@ static int snd_sof_fw_ext_man_parse(struct snd_sof_dev *sdev,
 	return ext_man_size;
 }
 
+static int snd_sof_fw_ext_man4_parse(struct snd_sof_dev *sdev, const struct firmware *fw)
+{
+	struct sof_ipc4_fw_modules *module_entry;
+	struct sof_man4_fw_binary_header *fw_header;
+	struct sof_ext_manifest4_hdr *hdr;
+	struct sof_man4_module_config *fm_config;
+	struct sof_man4_module_entry *fm_entry;
+	int fw_offset;
+	int i;
+
+	if (fw->size < sizeof(hdr)) {
+		dev_err(sdev->dev, "Invalid fw size\n");
+		return -EINVAL;
+	}
+
+	hdr = (struct sof_ext_manifest4_hdr *)fw->data;
+
+	fw_offset = hdr->len;
+
+	fw_header = (struct sof_man4_fw_binary_header *)(fw->data + fw_offset +
+							 SOF_MAN4_FW_HDR_OFFSET);
```

Comment:
This `0x2000` has been a constant for a long time: https://github.com/thesofproject/sof/commit/8885b465df53

To test https://github.com/thesofproject/sof/pull/4889 I ran the latest `tools/sof_ri_info/sof_ri_info.py` on all platform binaries since v1.7 and it parsed all of them. Example:

```
Reading SOF ri image sof-bin/v1.9.x/sof-v1.9/intel-signed/sof-tgl-h.ri
    File size 0x68300 (426752)
    0x0     Looking for Extended Manifest
    0x0     Extended Manifest (XMan)
    0x300   Looking for CSE Manifest
    0x300   CSE Manifest ($CPD)
    0x308   # of entries 3
    0x314   Reading CSE Manifest entry 0...
    0x32c   ... CSE Entry name ADSP.man file offset 0x35c length 1124
    ...
    0x2300  ADSP Manifest ($AM1)
    0x2334  Module Entry signature found ($AME)
    0x23a8  Module Entry signature found ($AME)
    Parsing finished

    SOF Binary sof-bin/v1.9.x/sof-v1.9/intel-signed/sof-tgl-h.ri size 0x68300

      Extended Manifest ver 1.0.0 length 768

      CSE Manifest ver 0x102 checksum 0x0 partition name ADSP

        ADSP.man (CSS Manifest) type 0x4 file offset 0x35c hdr_len 900 ver 0x21000 date 2021/09/23
          Rsvd0 0x1
          Modulus size (dwords) 96
            d3 72 92 99 4e b9 cd 67 ... 5d 62 4a 80 96 31 f8 b5 (TGL Intel prod key)
          Exponent size (dwords) 1
      ....
      ....

      cavs0015 (ADSP Manifest) file offset 0x2300 name ADSPFW build ver 1.9.0.1 feature mask 0xffff image flags 0x0
        HW buffers base address 0x0 length 0x0
        Load offset 0x30000


```